### PR TITLE
*: Add TTL to the self schema version on etcd

### PR DESCRIPTION
**Pull request by zimulala**
I will add tests later.

### CODE REVIEW

File: ddl/syncer.go [hunk 1]
```diff
@@ -44,9 +45,14 @@ const (
...
-// CheckVersFirstWaitTime is a waitting time before the owner checks all the servers of the schema version,
-// and it's an exported variable for testing.
-var CheckVersFirstWaitTime = 50 * time.Millisecond
+var (
+	// CheckVersFirstWaitTime is a waitting time before the owner checks all the servers of the schema version,
+	// and it's an exported variable for testing.
+	CheckVersFirstWaitTime = 50 * time.Millisecond
+	// SyncerSessionTTL is the etcd session's TTL in seconds.
+	// and it's an exported variable for testing.
+	SyncerSessionTTL = 30 * 60
```
Comment (tiancaiamao) [inline ddl/syncer.go]: Is this too large?

File: domain/domain.go [hunk 2]
```diff
@@ -310,11 +311,20 @@ func (do *Domain) loadSchemaInLoop(lease time.Duration) {
...
 				log.Errorf("[ddl] reload schema in loop err %v", errors.ErrorStack(err))
 			}
-		case <-do.ddl.SchemaSyncer().GlobalVersionCh():
+		case <-syncer.GlobalVersionCh():
 			err := do.Reload()
 			if err != nil {
 				log.Errorf("[ddl] reload schema in loop err %v", errors.ErrorStack(err))
 			}
+		case <-syncer.Done():
+			// The schema syncer stops, we need stop the schema validator to synchronize the schema version.
```
Comment (tiancaiamao) [inline domain/domain.go]: No, don't modify SchemaValidator.
It has nothing to do with here.

File: domain/schema_validator.go [hunk 3]
```diff
@@ -43,9 +49,29 @@ func newSchemaValidator(lease time.Duration) SchemaValidator {
 	}
 }
 
+func (s *schemaValidator) Stop() {
+	log.Infof("the schema validator stops")
```
Comment (shenli) [inline domain/schema_validator.go]: Infof -> info

File: domain/schema_validator.go [hunk 4]
```diff
@@ -43,9 +49,29 @@ func newSchemaValidator(lease time.Duration) SchemaValidator {
...
+	log.Infof("the schema validator restarts")
+	s.mux.Lock()
+	defer s.mux.Lock()
+	s.items = make(map[int64]time.Time)
+}
+
 func (s *schemaValidator) Update(leaseGrantTS uint64, schemaVer int64) {
 	s.mux.Lock()
 
+	if s.items == nil {
```
Comment (shenli) [inline domain/schema_validator.go]: Add defer s.mux.Unlock() before this if.

Comment (zimulala) [inline domain/schema_validator.go]: I will update later.

Comment (zimulala) [inline domain/schema_validator.go]: Done.

**Comment by zimulala** [comment]
PTAL @tiancaiamao @coocood @shenli

**Comment by tiancaiamao** [comment]
LGTM
please resolve conflicts.

**Comment by shenli** [comment]
LGTM
@coocood PTAL

**Pull request merged**